hail/src/main/scala/is/hail/expr/types/physical/PNDArray.scala

```diff
@@ -110,6 +110,52 @@ final case class PNDArray(elementType: PType, nDims: Int, override val required:
     )
   }
 
+  def linearizeIndices(indices: Array[Code[Long]], shapeArray: Array[Code[Long]], region: Code[Region], mb: MethodBuilder): Code[Long] = {
+    val index = mb.newField[Long]
+    val elementsInProcessedDimensions = mb.newField[Long]
+    Code(
+      index := 0L,
+      elementsInProcessedDimensions := 1L,
+      Code.foreach((shapeArray.length - 1) to 0 by - 1){ dimIndex =>
+        Code(
+          index := index + indices(dimIndex) * elementsInProcessedDimensions,
+          elementsInProcessedDimensions := elementsInProcessedDimensions * shapeArray(dimIndex)
+        )
+      },
+      index
+    )
+  }
+
+  def unlinearizeIndex(index: Code[Long], shapeArray: Array[Code[Long]], region: Code[Region], mb: MethodBuilder): (Code[Unit], Array[Code[Long]]) = {
+    val nDim = shapeArray.length
+
+    if (nDim <= 1) {
+      (Code._empty, Array(index))
+    }
+    else {
```

Comment:
`} else {`

Reply:
I have never liked the way the cuddled else looks, but sure.